src/bitbox02.js

```diff
@@ -86,25 +93,93 @@ const setOutputDefaults = outputs =>  {
 }
 
 export class BitBox02API {
+    /**
+     * @param devicePath See `getDevicePath()`.
+    */
     constructor(devicePath)  {
         this.devicePath = devicePath;
         this.opened = false;
-        this.socket = null;
+        // connection is an object with three keys once the connection is established:
+        // onWrite(bytes): send bytes
+        // close():  close the connection
+        // valid(): bool - is the connection still alive?
+        this.connection = null;
+        this.onCloseCb = null;
+
+        if (navigator.hid) {
+            navigator.hid.addEventListener("disconnect", () => {
```

Comment:
what if a user has 2 devices connected, one of which is bb02 and the other is something else. then they disconnect that something else: wouldn't it close the connection with bb02?

Reply:
It only fires if a device is disconnected that was previously connected with WebHID on the same page.

We only handle one connected device at the moment, so it should be okay.

The API here is really lacking, it is not obvious how to identify which device disconnected if you indeed had multiple. They should've done it like with the websocket, where the event is attached to the connection...

Reply:
:+1: